### PR TITLE
Make it possible to change Error-type

### PR DESCRIPTION
**Pull request by johlrogge**
I noticed a mistake in my try_filter_map_err-implementation that made it impossible to map to a different Error-type

### CODE REVIEW

File: src/try_filter_map.rs [hunk 1]
```diff
@@ -97,26 +97,25 @@ pub trait TryFilterMap<O, E>: Sized {
...
     where
-        F: FnMut(E) -> Option<Result<O, E>>;
+        F: FnMut(E) -> Option<Result<O, E2>>;
 }
 
 impl<I, O, E> TryFilterMap<O, E> for I
 where
     I: Iterator<Item = Result<O, E>> + Sized,
 {
-    #[inline]
```
Comment (matthiasbeyer) [inline src/try_filter_map.rs]: Why are you removing this?

File: src/try_filter_map.rs [hunk 2]
```diff
@@ -174,7 +173,8 @@ where
                     Some(r) => Some(r),
                     None => continue,
                 },
-                v => v,
+                Some(Ok(x)) => Some(Ok(x)),
+                None => None,
```
Comment (matthiasbeyer) [inline src/try_filter_map.rs:179]: Why was this changed? It doesn't make a difference, does it?

**Review by matthiasbeyer — CHANGES_REQUESTED** [review]

File: src/try_filter_map.rs (hunk 2 above)
Comment (johlrogge) [reply src/try_filter_map.rs:179]: It s subtle but `v` is a `Result<T, E>` but the function returns a `Result<T, E2>`.

File: src/try_filter_map.rs (hunk 1 above)
Comment (johlrogge) [reply src/try_filter_map.rs]: Sloppy mistake, I'll fix it

File: src/try_filter_map.rs (hunk 2 above)
Comment (matthiasbeyer) [reply src/try_filter_map.rs:179]: Ah. :laughing:

**Review by matthiasbeyer — APPROVED** [review]

**Comment by matthiasbeyer** [comment]
bors merge
